ecosystem/python/sdk/aptos_sdk/authenticator.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+import io
+import typing
+import unittest
+
+from account_address import AccountAddress
+from bcs import Deserializer, Serializer
+import ed25519
+
+
+class Authenticator:
+    """
+    Each transaction submitted to the Aptos blockchain contains a `TransactionAuthenticator`.
+    During transaction execution, the executor will check if every `AccountAuthenticator`'s
+    signature on the transaction hash is well-formed and whether the sha3 hash of the
+    `AccountAuthenticator`'s `AuthenticationKeyPreimage` matches the `AuthenticationKey` stored
+    under the participating signer's account address.
+    """
+
+    ED25519: int = 0
+    MULTI_ED25519: int = 1
+    MULTI_AGENT: int = 2
+
+    variant: int
+    authenticator: typing.Any
+
+    def __init__(self, authenticator: typing.Any):
+        if isinstance(authenticator, Ed25519Authenticator):
+            self.variant = 0
```

Comment:
You should be using the constants above ^

Reply:
fixed